packages/ui/components/auth.tsx

```diff
@@ -0,0 +1,95 @@
+import { AppConfig, UserSession, showConnect } from "@stacks/connect";
+// import { Storage } from '@stacks/storage';
+import { StacksMainnet, StacksTestnet } from "@stacks/network";
+import {
+  callReadOnlyFunction,
+  cvToJSON,
+  standardPrincipalCV,
+} from "@stacks/transactions";
+
+// Initialize Gaia hub permissions for the user
+const appConfig = new AppConfig(["store_write", "publish_data"]);
+
+// Set this to true if you want to use Mainnet
+// Using a boolean helps to switch the entire application to mainnet
+// or testnet with only changing a single value
+const isNetworkMainnet = true;
+
+export const userSession = new UserSession({ appConfig });
+// export const storage = new Storage({ userSession });
+
+// Return the network type object for the current network in use
+export function networkType() {
+  if (isNetworkMainnet) return new StacksMainnet();
+  else return new StacksTestnet();
+}
+
+export function getMyStxAddress(): string {
+  if (isNetworkMainnet) return getUserData().profile.stxAddress.mainnet;
+  else return getUserData().profile.stxAddress.testnet;
+}
+
+export function getNetworkName(): string {
+  if (isNetworkMainnet) return "mainnet";
+  else return "testnet";
+}
+
+export function authenticate(): void {
+  showConnect({
+    appDetails: {
+      name: "daoOS",
+      // TODO: add daoOS logo over here
+      icon: "../public/favicon.ico",
+    },
+    redirectTo: "/",
+    onFinish: () => {
+      // TODO: Enter miami vice's contract address
+      const contractAddress: string = "SP98329831323123";
+      // TODO: Enter miami vice's contract name
+      const contractName: string = "miami-vice-v1";
+
+      const functionName: string = "is-dao-member";
+      const stxAddress: string = getMyStxAddress();
+      const principalArg = standardPrincipalCV(stxAddress);
+
+      const options = {
+        contractAddress,
+        contractName,
+        functionName,
+        functionArgs: [principalArg],
+        network: networkType(),
+        senderAddress: stxAddress,
+      };
+
+      callReadOnlyFunction(options)
```

Comment:
I imagine this is untested since we don't have a testnet contract up yet. Could we mulit-line comment this call out until we're actually using it? This way we don't always have a POST error :)

Reply:
Right, makes sense xD